gcloud-java-storage/src/main/java/com/google/gcloud/spi/DefaultStorageRpc.java

```diff
@@ -465,15 +465,18 @@ public void write(String uploadId, byte[] toWrite, int toWriteOffset, long destO
       GenericUrl url = new GenericUrl(uploadId);
       HttpRequest httpRequest = storage.getRequestFactory().buildPutRequest(url,
           new ByteArrayContent(null, toWrite, toWriteOffset, length));
-      long limit = destOffset + length;
-      StringBuilder range = new StringBuilder("bytes ");
-      range.append(destOffset).append('-').append(limit - 1).append('/');
-      if (last) {
-        range.append(limit);
-      } else {
-        range.append('*');
+      // 0-length requests are valid, don't need content-range.
```

Comment:
Interesting, thanks for fixing this bug.
I assume you verified that you could write (and eventually finalize) after the 0 length write?
What happens if the 0 length write is the "last" call (and should finalize the file).
Also, from `gcloud-java` perspective do you think we should optimize the 0 length write and avoid the RPC (if not last)?